### PR TITLE
Feature/us12812 sync updated mp contact data to hubspot

### PR DESCRIPTION
**Pull request by agause**
Adds the sync-Ministry-Platform-contact-updates-to-HubSpot feature to the existing application.

### CODE REVIEW

File: Crossroads.Service.HubSpot.Sync.App/Controllers/SyncController.cs [hunk 1]
```diff
@@ -82,62 +100,72 @@ public IActionResult ResetJobProcessingState()
...
+                    _logger.LogError(AppEvent.Web.ViewLastSuccessfulSyncDates, exc, "An exception occurred viewing the sync processing dates.", exc);
                     throw;
                 }
             }
         }
 
-        [HttpGet]
-        [Route("dates/view")]
-        public IActionResult ViewLastSyncDates()
+        [HttpPost]
```
Comment (jpc11623) [inline Crossroads.Service.HubSpot.Sync.App/Controllers/SyncController.cs]: Should this be changed to something like date/{reg-sync-date} to follow route naming? (same for the other routes)

File: Crossroads.Service.HubSpot.Sync.App/Controllers/SyncController.cs [hunk 2]
```diff
@@ -61,6 +61,24 @@ public IActionResult SyncMpContactsToHubSpot()
...
+        {
+            using (_logger.BeginScope(AppEvent.Web.ViewJobProcessingState))
+            {
+                try
+                {
+                    return Content($"Current state: {_configurationService.GetCurrentJobProcessingState()}");
+                }
+                catch (Exception exc)
+                {
+                    _logger.LogError(AppEvent.Web.ViewJobProcessingState, exc, "An exception occurred viewing the sync processing state.", exc);
```
Comment (jpc11623) [inline Crossroads.Service.HubSpot.Sync.App/Controllers/SyncController.cs]: Will this log the name of the function the error occurred in?

File: Crossroads.Service.HubSpot.Sync.App/Controllers/SyncController.cs [hunk 3]
```diff
@@ -155,7 +183,7 @@ public IActionResult ViewActivity(string activityId)
         }
 
         [HttpGet]
-        [Route("activity/viewall")] // maybe create a view later
+        [Route("viewall")] // maybe create a view later
```
Comment (jpc11623) [inline Crossroads.Service.HubSpot.Sync.App/Controllers/SyncController.cs]: use standard routing syntax

File: Crossroads.Service.HubSpot.Sync.App/Startup.cs [hunk 4]
```diff
@@ -99,7 +102,7 @@ public void ConfigureServices(IServiceCollection services)
         public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
         {
             // Hackery for setting the application's base path for impending log files
-            log4net.GlobalContext.Properties["AppLogRoot"] = Environment.GetEnvironmentVariable("APP_LOG_ROOT");
+            log4net.GlobalContext.Properties["AppLogRoot"] = Configuration["APP_LOG_ROOT"];
```
Comment (jpc11623) [inline Crossroads.Service.HubSpot.Sync.App/Startup.cs]: the log4net config should have this available and pick it up from the env vars. Was that not working?

File: Crossroads.Service.HubSpot.Sync.App/Controllers/SyncController.cs (hunk 1 above)
Comment (agause) [reply Crossroads.Service.HubSpot.Sync.App/Controllers/SyncController.cs]: @jpc11623 Yes, when it's fully functional I'll make more meaningful routes. I'll remove it from the controller for now. Thanks!

File: Crossroads.Service.HubSpot.Sync.App/Controllers/SyncController.cs (hunk 2 above)
Comment (agause) [reply Crossroads.Service.HubSpot.Sync.App/Controllers/SyncController.cs]: @jpc11623 I think the idea behind including the EventId is to better distinguish when teams write to the Windows event log.

File: Crossroads.Service.HubSpot.Sync.App/Controllers/SyncController.cs (hunk 3 above)
Comment (agause) [reply Crossroads.Service.HubSpot.Sync.App/Controllers/SyncController.cs]: @jpc11623 I'll need some guidance on that. I'll bother you for it soon.

File: Crossroads.Service.HubSpot.Sync.App/Startup.cs (hunk 4 above)
Comment (agause) [reply Crossroads.Service.HubSpot.Sync.App/Startup.cs]: @jpc11623 I may have messed it up by doing so, but I deviated a bit from the way the micro service template prescribes setting up log4net. Perhaps we can look at that together.

**Pull request merged**